cornac/eval_methods/base_method.py

```diff
@@ -187,18 +188,30 @@ def pos_items(csr_row):
             else pos_items(train_mat.getrow(user_idx))
         )
 
-        u_gt_neg = np.ones(test_set.num_items, dtype='int')
-        u_gt_neg[test_pos_items + val_pos_items + train_pos_items] = 0
+        # binary mask for ground-truth negative items, removing all positive items
+        u_gt_neg_mask = np.ones(test_set.num_items, dtype="int")
+        u_gt_neg_mask[test_pos_items + val_pos_items + train_pos_items] = 0
+
+        # filter items being considered for evaluation
+        if exclude_unknowns:
+            item_indices = np.arange(train_set.num_items)
+            u_gt_pos_mask = u_gt_pos_mask[: train_set.num_items]
+            u_gt_neg_mask = u_gt_neg_mask[: train_set.num_items]
+        else:
+            item_indices = np.arange(test_set.num_items)
+        item_indices = np.nonzero(u_gt_pos_mask + u_gt_neg_mask)[0]
```

Comment:
Given this line, do we still need item_indices inside the above if - else block?